SofascoreAcademy/SofascoreAcademy/Controllers/MainViewController.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  MainViewController.swift
+//  SofascoreAcademy
+//
+//  Created by Matija Pavicic on 02.04.2024..
+//
+
+import Foundation
+import SnapKit
+import SofaAcademic
+import UIKit
+
+class MainViewController: UIViewController, BaseViewProtocol, TabItemDelegateProtocol {
+    
+    private let safeAreaCoverView: SafeAreaCoverView = .init()
+    private let mainHeaderView: MainHeaderView = .init()
+    private let eventsViewController: EventsViewController = .init()
+    private let settingsViewController: SettingsViewController = .init()
+    private let containerView: BaseView = .init()
+    private let tabView: TabView = .init()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tabView.delegate = self
+        
+        addViews()
+        setupConstraints()
+        styleViews()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        navigationController?.isNavigationBarHidden = true
+        
+        eventsViewController.dataIndex(UserDefaults.standard.integer(forKey: "tabIndex"))
```

Comment:
napravi neki UserDefaultsHelper ili PersistanceManager gdje ćeš raditi kontrolu nad UserDefaults, i tamo defeniraj key u string.

Možeš tamo imati ovako nešto
   static var example: Int {
        get { (UserDefaults.standard.integer(forKey: examleKey) as? Int ?? 0 }
        set { UserDefaults.standard.setValue(newValue, forKey: examleKey) }
    }

Reply:
ComputedProperty sa getterom i setterom